### DISCUSSION TITLE
How to connect to an existing irods server?

### DISCUSSION THREAD

**Discussion by alevikpes** [UtrechtUniversity/yoda#482] [Q&A] [answered]
Hi!

I have a real irods server running. Is it possible to connect to it using my current development setup? What would be the best way of doing it?

**Comment by stsnel** [comment]
Hi,

Do I understand correctly that you have a non-Yoda iRODS server and a Yoda development VM, and want to connect them so that you can manage data in one zone from the other one? 

iRODS itself supports federation of zones (https://docs.irods.org/master/system_overview/federation/). However, Yoda does not support this (more specifically: we don't use federation, testing federation scenario's is not part of our QA processes, and Yoda does not have any specific functionality for interacting with other zones).

If you can elaborate on your use case, perhaps we can provide additional advice.

**Reply by alevikpes** [reply]
The situation is like this, atm. We have an irods instance which is filled with real data and people use it via ssh or icommands. We want to make a frontend for it and we thought, it would be possible to use yoda for that purpose. As far as I understand, yoda has all included. So, my question now is it possible to adjust yoda to be as a frontend only? Another question is, in case we create a new full yoda instance, how difficult can be to transfer data from the existing irods to the new yoda-irods?

**Reply by stsnel** [reply, marked as answer]
Right, I see.

The Yoda frontend (the portal) depends on the backend (the ruleset and microservices), so it's not possible to run the portal on an iRODS server without also having the other Yoda components installed. In addition to that, the data will have to be organized according to Yoda's data and user structure. If you want to provide a graphical interface for an existing iRODS zone, you could perhaps consider the following alternatives:
- Metalnx (https://github.com/irods-contrib/metalnx-web) is a generic web interface for iRODS
- iBridges GUI (https://github.com/iBridges-for-iRODS/iBridges-GUI) is a local application that provides a graphical interface to iRODS.

If you want to migrate to Yoda, you will indeed need to transfer data objects / files to the Yoda servers.

In that case, it is probably a good idea to first design a group structure in Yoda. In Yoda, users are given access to data on a group level. It is therefore important that all data in a group can be shared with the same set of users. In practice, we see that data managers often create one or more groups for each (research) project. If a research project has multiple types of data (e.g. public data and sensitive data), it is generally stored in different groups. It's also a good idea to think about naming (e.g. URLs of the portal, zone name, etc.) at this stage, since such settings can be harder to change once you have active users and data on the system.

Once you have a group structure in Yoda, you could consider transferring data in one of the following ways:
1. Transfer data from the originating server or via an intermediate system using iRODS tooling (e.g. iCommands, GoCommands).
2. Connect the storage of the originating server to the Yoda system, adjusting paths so that they match the group structure and then registering the files using `ireg`. 

Getting this right will probably require doing a couple of test runs to see how these methods work in practice, to finetune the process and to estimate how much time you'll need to migrate the data.

**Reply by alevikpes** [reply]
Thank you for all this info.
